src/cursor/cur_hs.c

```diff
@@ -141,3 +141,100 @@ __wt_hs_cursor_search_near(WT_SESSION_IMPL *session, WT_CURSOR *cursor, int *exa
       session, WT_ISO_READ_UNCOMMITTED, ret = cursor->search_near(cursor, exactp));
     return (ret);
 }
+
+/*
+ * __curhs_close --
+ *     WT_CURSOR->close method for the hs cursor type.
+ */
+static int
+__curhs_close(WT_CURSOR *cursor)
+{
+    WT_CURSOR *file_cursor;
+    WT_CURSOR_HS *hsc;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    hsc = (WT_CURSOR_HS *)cursor;
+    file_cursor = hsc->file_cursor;
+    CURSOR_API_CALL_PREPARE_ALLOWED(
+      cursor, session, close, file_cursor == NULL ? NULL : CUR2BT(file_cursor));
+err:
+
+    if (file_cursor != NULL)
+        WT_TRET(file_cursor->close(file_cursor));
+    __wt_cursor_close(cursor);
+
+    API_END_RET(session, ret);
+}
+
+/*
+ * __curhs_reset --
+ *     Reset a history store cursor.
+ */
+static int
+__curhs_reset(WT_CURSOR *cursor)
+{
+    WT_CURSOR *file_cursor;
+    WT_CURSOR_HS *hsc;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    hsc = (WT_CURSOR_HS *)cursor;
+    file_cursor = hsc->file_cursor;
+    CURSOR_API_CALL_PREPARE_ALLOWED(cursor, session, reset, CUR2BT(file_cursor));
+
+    ret = file_cursor->reset(file_cursor);
+    F_CLR(cursor, WT_CURSTD_KEY_SET | WT_CURSTD_VALUE_SET);
+
+err:
+    API_END_RET(session, ret);
+}
+
+/*
+ * __wt_curhs_open --
+ *     Initialize a history store cursor.
+ */
+int
+__wt_curhs_open(WT_SESSION_IMPL *session, WT_CURSOR **cursorp)
```

Comment:
What is happening with `cursorp` when the function call is successful?

Reply:
Yeah, I missed that. Good catch.

Reply:
Fixed.